project_setup.py

```diff
@@ -0,0 +1,253 @@
+    not_image =  (
+        not file_path.endswith(".png")
+        and not file_path.endswith(".jpg")
+        and not file_path.endswith(".jpeg")
+    )
+    not_binary = not file_path.endswith(".DS_Store")
+    not_this = not file_path.endswith("project_setup.py")
+
+    return not_in_git and not_image and not_binary and not_this
+
+
+def get_repo_url(
+    file_contents:Iterable[str],
+    input_func:Callable[[Optional[str]], str] = input
+) -> str:
+    """Gets the URL of the remote repository.
+
+    Args:
+        file_contents (Iterable[str]): The contents of the .git/config file.
+        input_func (Callable[[Optional[str]], str], optional): The input function to use. Defaults to input.
+
+    Returns:
+        str: The URL of the remote repository.
+
+    """
+    # cp.read("./.git/config")
+    cp = configparser.ConfigParser()
+    try:
+        cp.read_file(file_contents)
+    except configparser.MissingSectionHeaderError:
+        url_guess = MSG_GIT_URL_NOT_FOUND
+
+    try:
+        url_guess = cp['remote "origin"']["url"]
+    except KeyError:
+        url_guess = MSG_GIT_URL_NOT_FOUND
+
+    # this could be a problem if someone accidentally hits enter and the URL is
+    # is set to MSG_GIT_URL_NOT_FOUND. Maybe change to a while loop if this
+    # becomes an issue.
+    repo_url = input_func(PROMPT_REMOTE_REPO.format(url_guess=url_guess)).strip()
+
+    return repo_url or url_guess
+
+
+def get_package_name(
+    repo_url:str,
+    input_func:Callable[[Optional[str]], str] = input
+) -> str:
+    """Gets the package name from the user.
+
+    Guesses the package name from the repo URL.
+
+    Args:
+        repo_url (str): The URL of the remote repository.
+        input_func (Callable[[Optional[str]], str], optional): The input function to use. Defaults to input.
+
+    Returns:
+        str: The package name.
+    """
+    repo_name = os.path.basename(repo_url).split('/')[-1].split('.')[0]
+    guessed_name = make_name_safe(repo_name)
+
+    input_name = input_func(PROMPT_PKG_NAME.format(name_guess=guessed_name)).strip()
+
+    return input_name or guessed_name
+
+
+def replace_file_contents(
+    new_str:str,
+    old_str:str,
+    file_path:str,
+) -> bool:
+    """Replaces instances of `old_str` with `new_str` in file at file_path.
+
+    Args:
+        old_str (str): The string to replace.
+        new_str (str): The string to replace `old_str` with.
+        file_path (str): The path to the file to modify.
+
+    Returns:
+        None
+    """
+    with open(file_path, "r") as f:
+        content = f.read()
+
+    new_content = content.replace(old_str, new_str)
+
+    if new_content != content:
+        with open(file_path, "w") as f:
+            f.write(new_content)
+        return True
+
+    return False
+
+
+def self_destruct(input_func:Callable[[str], str] = input) -> None:
+    """Deletes this file upon confirmation."""
+
+    if input_func("Delete this file? [y/N]: ").strip().lower() == "y":
```

Comment:
as above, nothing harmful in going ahead, and where probably most invocations defaults are OK

Reply:
sounds good.